atecc608a_se.c

```diff
@@ -211,6 +211,53 @@ static psa_status_t atecc608a_import_public_key(psa_key_slot_number_t key_slot,
     return status;
 }
 
+static psa_status_t atecc608a_generate_key(psa_key_slot_number_t key_slot,
+                                           psa_key_type_t type,
+                                           psa_key_usage_t usage,
+                                           size_t bits,
+                                           const void *extra,
+                                           size_t extra_size,
+                                           uint8_t *p_pubkey_out,
+                                           size_t pubkey_out_size,
+                                           size_t *p_pubkey_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+
+    /* The hardware has slots 0-15 */
+    if (key_slot > 15)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (type != PSA_KEY_TYPE_ECC_KEYPAIR(PSA_ECC_CURVE_SECP256R1))
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (bits != PSA_BYTES_TO_BITS(ATCA_PRIV_KEY_SIZE))
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (p_pubkey_out != NULL && pubkey_out_size < ATCA_PUB_KEY_SIZE)
```

Comment:
Is `pubkey_out_size` in user-facing pubkey format or internal hardware-facing pubkey format? Don't we need some sort of size translation here?

I assume `pubkey_out_size` is in user-facing pubkey format, which is different from the hardware format. Maybe the user-facing pubkey format requires a `0x4` on the front, and the size given includes that extra byte.

If the user-facing format and the hardware format do indeed occur, did we test that if the driver is called with a buffer that is large enough for the hardware format, but not large enough for the user-facing format, we fail as expected?

Reply:
That's an oversight on my part - this code lacks the format translation, that the public key exporting had. I'll add that, and as for the size testing - I'll add various tests as the first step towards interactive app refactoring.

Reply:
Updated.